ras_party/views/respondent_view.py

```diff
@@ -77,6 +77,18 @@ def _validate_get_respondent_params(ids, first_name, last_name, email):
                 raise BadRequest(f"'{party_id}' is not a valid UUID format for property 'id'")
 
 
+@respondent_view.route("/respondents/party_id/<party_id>", methods=["GET"])
+def get_respondent_by_party_id(party_id: UUID) -> Response:
+    if not is_valid_uuid4(party_id):
+        return make_response("party_id is not UUID", 400)
+
+    respondent = respondent_controller.get_respondent_by_party_id(party_id)
+    if respondent:
+        return make_response(respondent.to_dict(), 200)
+
+    return make_response(f"respondent not found for party_id {party_id}", 404)
```

Comment:
A none response from sql is perfectly fine in the controller, I only want to start using 404 response when it is returning, using werkzeug.exceptions http responses in the controller is slightly clumsy and should be avoided if possible. This is however a work in progress as this will need to be integrated with other uses of the code at some stage